providers/phuslog/logger.go

```diff
@@ -0,0 +1,49 @@
+package phuslog
+
+import (
+	"fmt"
+
+	"github.com/phuslu/log"
+
+	"github.com/grpc-ecosystem/go-grpc-middleware/v2/interceptors/logging"
+)
+
+// Compatibility check.
+var _ logging.Logger = &Logger{}
+
+// Logger is a phuslog logging adapter compatible with logging middlewares.
+type Logger struct {
+	log.GrcpGatewayLogger
+}
+
+// InterceptorLogger is a phuslog.Logger to Logger adapter.
+func InterceptorLogger(logger log.GrcpGatewayLogger) *Logger {
+	return &Logger{logger}
+}
+
+// Log implements the logging.Logger interface.
+func (l *Logger) Log(lvl logging.Level, msg string) {
+	switch lvl {
+	case logging.DEBUG:
+		l.Debug(msg)
+	case logging.INFO:
+		l.Info(msg)
+	case logging.WARNING:
+		l.Warning(msg)
+	case logging.ERROR:
+		l.Error(msg)
+	default:
+		// TODO(kb): Perhaps this should be a logged warning, defaulting to ERROR to get attention
+		// without interrupting code flow?
+		panic(fmt.Sprintf("phuslog: unknown level %s", lvl))
```

Comment:
Yeah I don't think this sounds reasonable, indeed maybe just fall back to error?

Reply:
```suggestion
		l.NoLevelErr(fmt.Errorf("phuslog: unknown level %s", lvl), msg)
```

I have just implemented a `NoLevelErr` which would output
```go
{"level":"error","error":"phuslog: unknown level AAA","message":"a message log"}
```

Given that we are logging with unknown level I thinks it's best to choose Error Level because is the minimum any user would have...

If you agree I will push changes algo to phuslog

Reply:
Only change this one, and yes fall back to error.